packages/gen-ai-hub/src/client/openai/openai-client.ts

```diff
@@ -1,60 +1,107 @@
+   * @param model - The model to use for the chat completion.
    * @param data - The input parameters for the chat completion.
+   * @param deploymentResolver - A deployment id or a function to retrieve it.
    * @param requestConfig - The request configuration.
    * @returns The completion result.
    */
   async chatCompletion(
+    model: OpenAiChatModel | { name: OpenAiChatModel; version: string },
     data: OpenAiChatCompletionParameters,
+    deploymentResolver?: DeploymentResolver,
     requestConfig?: CustomRequestConfig
   ): Promise<OpenAiChatCompletionOutput> {
+    const deploymentId = await resolveOpenAiDeployment(
+      model,
+      deploymentResolver
+    );
     const response = await executeRequest(
       {
-        url: `/inference/deployments/${data.deploymentConfiguration.deploymentId}/chat/completions`,
+        url: `/inference/deployments/${deploymentId}/chat/completions`,
         apiVersion
       },
       data,
-      requestConfig
+      mergeRequestConfig(requestConfig)
     );
     return response.data;
   }
   /**
    * Creates an embedding vector representing the given text.
-   * @param data - The input parameters for the chat completion.
+   * @param model - The model to use for the embedding computation.
+   * @param data - The text to embed.
+   * @param deploymentResolver - A deployment id or a function to retrieve it.
    * @param requestConfig - The request configuration.
    * @returns The completion result.
    */
   async embeddings(
+    model:
+      | OpenAiEmbeddingModel
+      | { name: OpenAiEmbeddingModel; version: string },
     data: OpenAiEmbeddingParameters,
+    deploymentResolver?: DeploymentResolver,
     requestConfig?: CustomRequestConfig
   ): Promise<OpenAiEmbeddingOutput> {
+    const deploymentId = await resolveOpenAiDeployment(
+      model,
+      deploymentResolver
+    );
     const response = await executeRequest(
-      {
-        url: `/inference/deployments/${data.deploymentConfiguration.deploymentId}/embeddings`,
-        apiVersion
-      },
+      { url: `/inference/deployments/${deploymentId}/embeddings`, apiVersion },
       data,
-      requestConfig
+      mergeRequestConfig(requestConfig)
     );
     return response.data;
   }
 }
+
+async function resolveOpenAiDeployment(
+  model: string | { name: string; version: string },
+  resolver?: DeploymentResolver
+) {
+  if (typeof resolver === 'string') {
+    return resolver;
+  }
+  const llm =
+    typeof model === 'string' ? { name: model, version: 'latest' } : model;
+  const deployment = await resolveDeployment({
+    scenarioId: 'foundation-models',
+    executableId: 'azure-openai',
+    model: llm
+  });
+  return deployment.id;
+}
+
+function mergeRequestConfig(
+  requestConfig?: CustomRequestConfig
+): HttpRequestConfig {
+  return {
+    method: 'POST',
+    headers: {
+      'content-type': 'application/json'
+    },
+    params: { 'api-version': apiVersion },
+    ...requestConfig
```

Comment:
here the requestConfig will entirely replace the headers and params keys.

Reply:
yes, this is intended. My thinking was that the open AI client should only mix in whatever is specific to the open ai client.